### PR TITLE
[NeMo-UX] proposal for updated 2.0 progress logger

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?

This introduces a new `MegatronProgressPrinter` class which simply prints the metrics to stdout in a similar style to Megatron. This implementation is more robust as the TQDM progress bar is not well-suited for non-interactive settings.

The train step logging looks like this: 
```
Epoch 0, iteration 1/100: | global_step: 0 | reduced_train_loss: 10.93 | train_step_timing in s: 1.352
Epoch 0, iteration 2/100: | global_step: 1 | reduced_train_loss: 9.409 | train_step_timing in s: 1.016
Epoch 0, iteration 3/100: | global_step: 2 | reduced_train_loss: 9.619 | train_step_timing in s: 1.018
Epoch 0, iteration 4/100: | global_step: 3 | reduced_train_loss: 8.796 | train_step_timing in s: 1.014
```

And the validation step logging looks like this:
```
Validation: batch 1 / 32
Validation: batch 2 / 32
Validation: batch 3 / 32
Validation: batch 4 / 32
```

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/trainer.py [hunk 1]
```diff
@@ -7,9 +8,35 @@
 from nemo.lightning.fabric.conversion import to_fabric
 from nemo.lightning.fabric.fabric import Fabric
 from nemo.lightning.io.mixin import IOMixin, serialization, track_io
+from nemo.lightning.pytorch.callbacks import MegatronProgressPrinter
 
 
 class Trainer(pl.Trainer, IOMixin):
+    def __init__(
```
Comment (marcromeyn) [inline nemo/lightning/pytorch/trainer.py:15]: Let's change this design. I really would prefer not to overwrite the `__init__` here since we loose auto-complete. 

Since it's megatron-specific, it can be moved inside the `MegatronStrategy`. It can be put inside [this](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/strategies.py#L241) method.

**Review by marcromeyn — CHANGES_REQUESTED** [review]

**Review by ShriyaPalsamudram — DISMISSED** [review]
LGTM, thanks

**Review by ShriyaPalsamudram — DISMISSED** [review]
LGTM, thanks

File: nemo/lightning/pytorch/strategies.py [hunk 2]
```diff
@@ -151,6 +155,8 @@ def __init__(
         ckpt_load_directly_on_device: bool = True,
         setup_optimizers: bool = True,
         init_model_parallel: bool = True,
+        replace_progress_bar: bool = True,
+        progress_interval: int = 100,
```
Comment (hemildesai) [inline nemo/lightning/pytorch/strategies.py:159]: Can we update this to 1?

**Review by hemildesai — APPROVED** [review]
